src/app/tasks/create-task/create-task.component.ts

```diff
@@ -1,44 +1,60 @@
-import { Component, EventEmitter, Output } from '@angular/core';
-import { Task } from '../task.model';
-import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { Component, EventEmitter, OnInit, Output, Input } from '@angular/core';
+import { Task, Option, OptionIndex } from '../task.model';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
 import { TaskService } from '../task.service';
-import { Validators } from '@angular/forms';
+import { KeyValuePipe } from '@angular/common';
 
 @Component({
   selector: 'app-create-task',
   standalone: true,
   imports: [
-    ReactiveFormsModule
+    ReactiveFormsModule,
+    KeyValuePipe
   ],
   templateUrl: './create-task.component.html',
   styleUrl: './create-task.component.scss'
 })
-export class CreateTaskComponent {
-  
+export class CreateTaskComponent implements OnInit{
+ 
   @Output() taskCreatedEvent = new EventEmitter<Task>();
 
+  @Input() taskTypes : Option[] = [];
+
+  @Input() taskStatuses : Option[] = [];
```

Comment:
Yeah here you have Option array, where you can assign the types and statuses from the taskService

Reply:
While in the TaskService I needed the Types and Statuses to be indexed so I can select a specific one, I needed to put them into arrays because I couldn't iterate over the indexed options in the template